lib/compilers/lfortran.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2022, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import type {ParseFiltersAndOutputOptions} from '../../types/features/filters.interfaces.js';
+
+import {FortranCompiler} from './fortran.js';
+
+export class LFortranCompiler extends FortranCompiler {
+    static override get key() {
+        return 'lfortran';
+    }
+
+    override optionsForFilter(filters: ParseFiltersAndOutputOptions, outputFilename: string) {
+        // TODO add `-g` to the options
```

Comment:
Who is this TODO fo? Why not add `-g` now?

Reply:
This TODO is for me, I will try to fix the issue related to enable -g option next week.
I just wanted to LFortran to be live. After that I will test everything and submit another PR to fix the issues.

Problems
- LFortran runtime stacktrace is dependent on llvm-dwarfdump, but while I ran the local instance I was not able to link it. I used `tools.llvm-dwarfdumptrunk.exe=<path>/bin/llvm-dwarfdump` with all other options. But, lfortran is not able to find it.
- I need to disable `-g` option to disable runtime stacktrace. 
- I'm running the local instance on MacOS and I get:
   ```
   Could not execute the program 
   No execution available for amd64-linux-cpu
   ``` 
   in "Execution Only", so I'm not sure if the output works

I will try to fix this issues after LFortran is live.

Reply:
Understood. In general having "TODO" with no context is unhelpful to other people. Like "TODO x" ... why not just do "x" ... so that's why I ask. I'll update this in the PR when I add the fortran compilers.